wasm/src/peerconnection.cpp

```diff
@@ -77,10 +101,31 @@ PeerConnection::PeerConnection(const Configuration &config) {
 	rtcSetDataChannelCallback(mId, DataChannelCallback);
 	rtcSetLocalDescriptionCallback(mId, DescriptionCallback);
 	rtcSetLocalCandidateCallback(mId, CandidateCallback);
+	rtcSetStateChangeCallback(mId, StateChangeCallback);
+	rtcSetGatheringStateChangeCallback(mId, GatheringStateChangeCallback);
+	rtcSetSignalingStateChangeCallback(mId, SignalingStateChangeCallback);
 }
 
 PeerConnection::~PeerConnection() { rtcDeletePeerConnection(mId); }
 
+optional<Description> PeerConnection::localDescription() const {
+	// TODO: sdp and type of the local description in acquired by two function calls.
+	// Find a way to reduce this into one call (probably multi-value return may work).
```

Comment:
It looks good to me this way. The C API of libdatachannel works like that. Because of how the API is designed, the local description should not change between calls.